process/track/baseBlockTrack.go

```diff
@@ -518,7 +518,20 @@ func (bbt *baseBlockTrack) GetTrackedHeadersWithNonce(shardID uint32, nonce uint
 // IsShardStuck returns true if the given shard is stuck
 func (bbt *baseBlockTrack) IsShardStuck(shardID uint32) bool {
 	numPendingMiniBlocks := bbt.blockBalancer.GetNumPendingMiniBlocks(shardID)
-	isShardStuck := numPendingMiniBlocks >= process.MaxNumPendingMiniBlocks
+	shardProcessedMetaNonce := bbt.blockBalancer.GetLastShardProcessedMetaNonce(shardID)
+
+	isMetaDifferenceTooLarge := false
+	if shardID != core.MetachainShardId {
+		lastMetaHeader, _, err := bbt.GetLastCrossNotarizedHeader(core.MetachainShardId)
```

Comment:
I think that GetLastCrossNotarizedHeader should be replaced with GetTrackedHeaders, for two reasons: 1) First method will return last cross notarized meta header from the node point of view and is related to where the node reached with the processed meta headers (so if this shard also has not kept the sync with meta it would consider that another shard which is closer with processing to the same meta header to be also synced which is false) 2) If IsShardStuck method would be called by metachain when it creates miniblocks for shards, like rewards or maybe other future miniblocks type, and it wants to check if the given shard is stuck or not, the call of method GetLastCrossNotarizedHeader(core.MetachainShardId) it would return only the genesis meta header, as this notarization does not exists (metachain will store cross notarization only for shards but not for himself. For own shard notarization it is used GetLastSelfNotarizedHeader method which is backed by another map)

Reply:
So I propose that lines between 525-531 to be replaced with this code:

		metaHeaders, _ := bbt.GetTrackedHeaders(core.MetachainShardId)
		numMetaHeaders := len(metaHeaders)
		if numMetaHeaders > 0 {
			lastMetaHeader := metaHeaders[numMetaHeaders - 1]
			metaDiff := lastMetaHeader.GetNonce() - shardProcessedMetaNonce
			isMetaDifferenceTooLarge = metaDiff > process.MaxMetaNoncesBehind
		}

Reply:
done